controller/internal/routes/ca_api_model.go

```diff
@@ -18,119 +18,103 @@ package routes
+	BasicLinkFactory
 }
 
-type CaApiCreate struct {
-	Tags                      map[string]interface{} `json:"tags"`
-	Name                      *string                `json:"name"`
-	CertPem                   *string                `json:"certPem"`
-	IsAutoCaEnrollmentEnabled *bool                  `json:"isAutoCaEnrollmentEnabled"`
-	IsOttCaEnrollmentEnabled  *bool                  `json:"isOttCaEnrollmentEnabled"`
-	IsAuthEnabled             *bool                  `json:"isAuthEnabled"`
-	IdentityRoles             []string               `json:"identityRoles"`
-	IdentityNameFormat        *string                `json:"identityNameFormat"`
+func NewCaLinkFactory() *CaLinkFactoryImpl {
+	return &CaLinkFactoryImpl{
+		BasicLinkFactory: *NewBasicLinkFactory(EntityNameCa),
+	}
 }
 
-func (i *CaApiCreate) ToModel() *model.Ca {
-	result := &model.Ca{}
-	result.Name = stringz.OrEmpty(i.Name)
-	result.Tags = i.Tags
-	result.CertPem = stringz.OrEmpty(i.CertPem)
-	result.IsAutoCaEnrollmentEnabled = i.IsAutoCaEnrollmentEnabled != nil && *i.IsAutoCaEnrollmentEnabled
-	result.IsOttCaEnrollmentEnabled = i.IsOttCaEnrollmentEnabled != nil && *i.IsOttCaEnrollmentEnabled
-	result.IsAuthEnabled = i.IsAuthEnabled != nil && *i.IsAuthEnabled
-	result.IdentityRoles = i.IdentityRoles
-	result.IdentityNameFormat = stringz.OrEmpty(i.IdentityNameFormat)
-	return result
-}
+func (factory *CaLinkFactoryImpl) Links(entity models.Entity) rest_model.Links {
+	ca := entity.(*model.Ca)
 
-type CaApiList struct {
-	*env.BaseApi
-	Name                      *string  `json:"name"`
-	Fingerprint               *string  `json:"fingerprint"`
-	CertPem                   *string  `json:"certPem"`
-	IsVerified                *bool    `json:"isVerified"`
-	VerificationToken         *string  `json:"verificationToken"`
-	IsAutoCaEnrollmentEnabled *bool    `json:"isAutoCaEnrollmentEnabled"`
-	IsOttCaEnrollmentEnabled  *bool    `json:"isOttCaEnrollmentEnabled"`
-	IsAuthEnabled             *bool    `json:"isAuthEnabled"`
-	IdentityRoles             []string `json:"identityRoles"`
-	IdentityNameFormat        *string  `json:"identityNameFormat"`
-}
+	links := factory.BasicLinkFactory.Links(entity)
+	if ca != nil {
+		if !ca.IsVerified {
+			links["verify"] = factory.NewNestedLink(entity, "verify")
+		}
 
-func (CaApiList) BuildSelfLink(id string) *response.Link {
-	return response.NewLink(fmt.Sprintf("./%s/%s", EntityNameCa, id))
-}
+		if ca.IsAutoCaEnrollmentEnabled {
+			links["jwt"] = factory.NewNestedLink(entity, "jwt")
+		}
+	}
 
-func (e *CaApiList) GetSelfLink() *response.Link {
-	return e.BuildSelfLink(e.Id)
+	return links
 }
 
-func (e *CaApiList) PopulateLinks() {
-	if e.Links == nil {
-		e.Links = &response.Links{
-			EntityNameSelf: e.GetSelfLink(),
-		}
+func MapCreateCaToModel(ca *rest_model.CaCreate) *model.Ca {
+	ret := &model.Ca{
+		BaseEntity: models.BaseEntity{
+			Tags: ca.Tags,
+		},
+		Name:                      stringz.OrEmpty(ca.Name),
+		Fingerprint:               "",
```

Comment:
i bet there is a reason why no fingerprint? maybe a comment would allow me to understand it. my instinct would be to fill it out

Reply:
It is handled at a lower level. 

The REST API `Map*` functions are largely meant to be dumb and defer to lower levels for logic and validation that can't be done on the REST API payload alone. Putting logic that logic here means any other presentation we use for the API (i.e. GRPC, channel2, etc.) would also have to have that logic.